Add help message & icon to empty automation editor

AW1534 opened the pull request:
This PR brings the changes from #8148 into the automation editor

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/b8d2d468-0954-4934-9196-c781367bc064" />

regulus79 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -1309,9 +1308,6 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
-	{
 		//NEEDS Change in CSS
 		//int len_ticks = 4;
 		timeMap & time_map = m_clip->getTimeMap();
```
Does removing this `if` statement leave the block with an extra indent?

allejok96 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -215,7 +214,11 @@ void AutomationEditor::updateAfterClipChange()
 	m_currentPosition = 0;
 
-	if( !validClip() )
+	setEnabled(validClip());
```
setEnabled should happen in AutomationEditorWindow to disable the toolbar buttons

allejok96 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -1055,58 +1058,55 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 
 	p.fillRect(0, TOP_MARGIN, VALUES_WIDTH, height() - TOP_MARGIN, m_scaleColor);
```
I think this is the part that draw the gray areas, you might want to move the if statement above this

allejok96 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -1055,58 +1058,55 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 	auto text_flags = (Qt::Alignment)(Qt::AlignRight | Qt::AlignVCenter);
 
-	if( validClip() )
+	if (m_y_auto)
```
I wouldn't touch this part of the code, just to keep the commit to the point. But that's a personal opinion.

allejok96 replied:
Ok other devs think this is better, so I withdraw my opinion

allejok96 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -2095,7 +2052,7 @@ AutomationEditorWindow::AutomationEditorWindow() :
 
 	// Interpolation actions
-	DropToolBar *interpolationActionsToolBar = addDropToolBarToTop(tr("Interpolation controls"));
+	m_editor->m_interpolationActionsToolBar = addDropToolBarToTop(tr("Interpolation controls"));
```
You could perhaps do something like this instead of hiding the toolbar
```cpp
connect(m_editor, &AutomationEditor::currentClipChanged, this, [this]{ setEnabled(m_editor->hasClip()); });
```

AW1534 commented:
> It should be fixed in `AutomationEditorWindow::updateEditTanButton` and not by hiding the button, because what if a key-combo activated it?

I've added a `validClip()` check in `updateEditTanButton()`, and that fixed the problem alone. but I've also taken your suggestion to disable the editor with a signal when `validClip()` returns false. I've changed this in the Piano Roll too.

regulus79 commented on src/gui/editors/PianoRoll.cpp:
```diff
@@ -4913,6 +4913,9 @@ PianoRollWindow::PianoRollWindow() :
 
+	// disable all controls when no clip is selected
+	connect(m_editor, &PianoRoll::currentMidiClipChanged, this, [this]{ setEnabled(m_editor->hasValidMidiClip()); });
+
```
It appears to still be there for me(?) https://github.com/AW1534/lmms/blob/429957864d5fe19d5fbdf8f9701307ac7aec270a/src/gui/editors/PianoRoll.cpp#L5469

AW1534 replied:
oh weird, its not there on my end and i have no uncommitted/unpushed changes

AW1534 replied:
Okay, I'll revert all my Piano Roll changes

regulus79 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -28,39 +28,39 @@
 #include "AutomationEditor.h"
 
 #include <QApplication>
+#include <QDebug>
```
I assume this `QDebug` include was intended to be removed?

regulus79 commented on src/gui/editors/AutomationEditor.cpp:
```diff
@@ -28,39 +28,39 @@
 #include <QScrollBar>
 #include <QStyleOption>
 #include <QToolTip>
+#include <SongEditor.h>
```
What is the purpose of this include? Are you now accessing the Song Editor somewhere here? Also why is it using `< >` instead of `" "`

AW1534 replied:
Unsure, my IDE must have included it

AW1534 commented:
Should be good for merge
